test/clojure_lsp/handlers_test.clj

```diff
@@ -55,6 +55,81 @@
       (is (= "xx/bar" (get-in changes ["file://b.clj" 1 :new-text]))))))
 
 (deftest test-find-diagnostics
+  (testing "wrong arity"
+    (testing "for argument destructuring"
+      (reset! db/db {:file-envs {"file://a.clj" (parser/find-usages "(defn foo ([x] x) ([x y] (x y)))
+                                                                     (defn bar [y & rest] ((foo y y y) (bar rest)))
+                                                                     (defn baz [{x :x y :y :as long}
+                                                                                {:keys [k v] :as short}
+                                                                                [_ a b]]
+                                                                       (x y k v a b long short))
+                                                                     (baz :broken :brokken [nil :ok :okay])
+                                                                     (baz {bar baz foo :no?})
+                                                                     (bar)
+                                                                     (bar {:a [:b]})
+                                                                     (bar :one-fish :two-fish :red-fish :blue-fish)
+                                                                     [foo]
+                                                                     {foo 1 2 3}
+                                                                     [foo 1 (foo 5 6 7)]
+                                                                     (foo)
+                                                                     (foo 1)
+                                                                     (foo 1 ['a 'b])
+                                                                     (foo 1 2 3 {:k 1 :v 2})" :clj {})}})
+      (let [usages (crawler/find-diagnostics #{} "file://a.clj" (get-in @db/db [:file-envs "file://a.clj"]))]
+        (is (= ["No overload foo for 3 arguments"
+                "No overload baz for 1 argument"
+                "No overload bar for 0 arguments"
+                "No overload foo for 3 arguments"
+                "No overload foo for 0 arguments"
+                "No overload foo for 4 arguments"]
+               (map :message usages)))))
+    (testing "for threading macros"
+      (reset! db/db {:file-envs {"file://a.clj" (parser/find-usages "(defn foo ([x] x) ([x y z] (z x y)))
+                                                                     (defn bar [] :bar)
+                                                                     (defn baz [arg & rest] (apply arg rest))
+                                                                     (->> :test
+                                                                       (foo)
+                                                                       (foo 1)
+                                                                       (bar))
```

Comment:
There's an issue with thread macros when you don't wrap single arity functions in parens. If you change this line to just `bar` you'll notice the diagnostic disappears. If you look at the `find-usages` result  you'll see the reference is missing.

I suspect parser test code like `(defn x [y] y) (-> 1 y)` will show you a difference compared `(defn x [y] y) (-> 1 (y))`.

Reply:
Okay I think this is fixed now, I added a parser test to check `:argc` values on usages as well as adding a case of this in the handler test.